lib/mongoid/extensions/float.rb

```diff
@@ -36,10 +36,8 @@ module ClassMethods
         #
         # @return [ String ] The object mongoized.
         def mongoize(object)
-          unless object.blank?
-            __numeric__(object).to_f rescue 0.0
-          else
-            nil
+          wrap_mongoize(object) do
+            __numeric__(object).to_f rescue nil
```

Comment:
I made the decision to make "" an uncastable value, and to raise instead of just returning nil, which was old functionality.

Reply:
Oleg and I agreed that "" should always return nil.